Studio-Lovelies/GG-JointJustice-Unity
# Refactor/appearing text controller

## Summary

Rewrites AppearingDialogueController class to make it simpler and shorter, and adds associated tests.
Also restructures the SpeechPanel game object to make better use of Unity UI.

## Testing instructions

Run the tests, and check the methods to make sure no required functionality is lost.

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
